syntax = "proto3";

service ServerExchange {
    rpc SyncDB (SyncDBRequest) returns (SyncDBResponse) {}
    rpc GetLogicalClock (GetLogicalClockRequest) returns (GetLogicalClockResponse) {}
}

message GetLogicalClockRequest {
}

message GetLogicalClockResponse {
    int32 logical_clock = 1;
}

message DBAccount {
    string username = 1;
    int32 logged_in = 2;
    string created_at = 3;
  }

  message DBMessage {
    string id = 1;
    string sender_username = 2;
    string reciever_username = 3;
    string content = 4;
    int32 delivered = 5;
    string created_at = 6;
  }


message SyncDBRequest {
    string server_address = 1;
    int32 server_port = 2;
    string timestamp = 3;
}

message SyncDBResponse {
    string timestamp = 1;
    repeated DBAccount db_accounts = 2;
    repeated DBMessage db_messages = 3;
}